### PR TITLE
Tiling angle support

### PR DESCRIPTION
**Pull request by BiancoA**
following back from issue: https://github.com/lovell/sharp/issues/1115

I exposed the angle argument for the .tile() function. It didn't really solve my problem, but it rotates the image. It looks equivalent to do .rotate() before .tile()

### CODE REVIEW

File: src/pipeline.cc [hunk 1]
```diff
@@ -1263,6 +1264,7 @@ NAN_METHOD(pipeline) {
   baton->tileSize = AttrTo<uint32_t>(options, "tileSize");
   baton->tileOverlap = AttrTo<uint32_t>(options, "tileOverlap");
   std::string tileContainer = AttrAsStr(options, "tileContainer");
+  baton->tileAngle = AttrTo<uint32_t>(options, "tileAngle");
```
Comment (lovell) [inline src/pipeline.cc]: The use of `uint32_t` here will discard negative values. `int32_t` should suffice.

File: test/unit/tile.js [hunk 2]
```diff
@@ -146,13 +146,38 @@ describe('Tile', function () {
 
   it('Prevent larger overlap than default size', function () {
     assert.throws(function () {
-      sharp().tile({overlap: 257});
+      sharp().tile({
```
Comment (lovell) [inline test/unit/tile.js]: I notice there are code layout changes unrelated to this PR. Is this using `prettier`? (Not a problem, more for interest.)

File: test/unit/tile.js [hunk 3]
```diff
@@ -192,6 +217,26 @@ describe('Tile', function () {
...
+        })
+        .toFile(fixtures.path('output.512_90.dzi'), function (err, info) {
+          if (err) throw err;
+          assert.strictEqual('dz', info.format);
+          assert.strictEqual(2725, info.width);
+          assert.strictEqual(2225, info.height);
+          assert.strictEqual(3, info.channels);
+          assert.strictEqual('undefined', typeof info.size);
+          assertDeepZoomTiles(directory, 512, 13, done);
+        });
```
Comment (lovell) [inline test/unit/tile.js]: I wonder if there's a reliable way to test that the tiles have actually been rotated by 90 degrees? Perhaps inspect one tile against a known fixture?

File: src/pipeline.cc (hunk 1 above)
Comment (BiancoA) [reply src/pipeline.cc]: fixed! Sorry for the headless copy-paste

File: test/unit/tile.js (hunk 2 above)
Comment (BiancoA) [reply test/unit/tile.js]: I use atom editor and this is the result of Beautify. I like it, but If you prefer to be consistent with the other convention you were using I can revert these changes.

File: test/unit/tile.js (hunk 3 above)
Comment (BiancoA) [reply test/unit/tile.js]: this may work, what do you think?

**Comment by lovell** [comment]
Brilliant, thanks for the updates Andrea. This feature will be included in the forthcoming v0.19.1.

**Pull request merged**